### PR TITLE
LibCore+Everywhere: Remove a few landmines

### PR DESCRIPTION
**Pull request by BenWiederhake**
This PR addresses some landmines; I found them by blowing my leg off (i.e. writing code that accidentally misuses something):
- All derivatives of `Core::Object` should have `protected` or `private` constructors, and only be created through `SomeClass::construct(…)`, as these are ref-counted objects, and handling the Object directly without ever creating a `RefPtr` has weird effects (e.g. if some other code decides to create a new RefPtr and later drop it, suddenly the destructor of the object gets called and the memory freed. Whoops!)
- The `Object::did_construct()` automagic mechanism was used only in one place (`WindowServer::Window`), where it was not necessary. Also, post-constructor initialization is a bit of a wart, let's not encourage it.
- Some drive-by fixes to mark overriding methods as `override`.

### CODE REVIEW

File: Userland/Applications/SoundPlayer/PlaylistWidget.h [hunk 1]
```diff
@@ -40,19 +40,22 @@ class PlaylistTableView : public GUI::TableView {
     void doubleclick_event(GUI::MouseEvent& event) override;
 
     Function<void(const Gfx::Point<int>&)> on_doubleclick;
+
+private:
+    PlaylistTableView() = default;
```
Comment (PeterBindels-TomTom) [inline Userland/Applications/SoundPlayer/PlaylistWidget.h:45]: Move to implementation file; this creates 3 extra symbol references in object files including this header.

File: Userland/Libraries/LibGUI/Calendar.h [hunk 2]
```diff
@@ -75,12 +72,15 @@ class Calendar final : public GUI::Frame {
...
+    Calendar(Core::DateTime date_time = Core::DateTime::now(), Mode mode = Month);
+    virtual ~Calendar() override;
+
     virtual void resize_event(GUI::ResizeEvent&) override;
     virtual void paint_event(GUI::PaintEvent&) override;
     virtual void mousemove_event(GUI::MouseEvent&) override;
     virtual void mousedown_event(MouseEvent&) override;
     virtual void mouseup_event(MouseEvent&) override;
-    virtual void doubleclick_event(MouseEvent&);
+    virtual void doubleclick_event(MouseEvent&) override;
```
Comment (PeterBindels-TomTom) [inline Userland/Libraries/LibGUI/Calendar.h:83]: Did this not trigger a warning? We should check that it does.

Comment (BenWiederhake) [reply Userland/Libraries/LibGUI/Calendar.h:83]: It did not. I only did these fixes as a drive-by, so I don't really want to write a new check for this.

**Comment by BenWiederhake** [comment]
Changes since last push:
- Move the PlaylistTableView constructor to its own file (Thanks @PeterBindels-TomTom!)
- Rebased for convenience

**Review by sin-ack — APPROVED** [review]

**Pull request merged**